dashboard/app/assets/javascripts/soundEffects.js

```diff
@@ -174,6 +188,17 @@ Sound.prototype.play = function (options) {
 
   this.audioElement.volume = typeof options.volume === "undefined" ? 1 : options.volume;
   this.audioElement.loop = !!options.loop;
+  if (options.onEnded) {
+    var unregisterAndCallback = function () {
+      this.audioElement.removeEventListener('abort', unregisterAndCallback);
+      this.audioElement.removeEventListener('ended', unregisterAndCallback);
+      this.audioElement.removeEventListener('pause', unregisterAndCallback);
+      options.onEnded();
+    }.bind(this);
+    this.audioElement.addEventListener('abort', unregisterAndCallback);
+    this.audioElement.addEventListener('ended', unregisterAndCallback);
+    this.audioElement.addEventListener('pause', unregisterAndCallback);
```

Comment:
We have to hook up a bunch of fallback player events.
* 'abort' fires when something unexpected interrupts playback (not when we stop it - this event is rare).
* 'ended' fires when the sound reaches the end of playback on its own - which is _never_ for looping sounds.
* 'pause' fires when playback is paused, which is the command we actually issue when we call `Sound.stop()` using the fallback player.